goon/browserassets/js/browserOutput.js

```diff
@@ -157,13 +158,15 @@ function setHighlightColor(match) {
 
 //Highlights words based on user settings
 function highlightTerms(el) {
-	var element = $(el)
-	if(!(element.mark)) { // mark.js isn't loaded; give up
-		return
-	}
 	for (var i = 0; i < opts.highlightTerms.length; i++) { //Each highlight term
 		if(opts.highlightTerms[i]) {
-			element.mark(opts.highlightTerms[i], {"element" : "span", "each" : setHighlightColor});
+			var rexp = new RegExp(opts.highlightTerms[i],"gmi")
+			if(!opts.highlightRegexEnable){
+				rexp = new RegExp("("+opts.highlightTerms[i].replace(new RegExp("([^a-zA-Z0-9])","gmi"),"\\$1")+")","gmi")
+				el.innerText = el.innerText.replace(rexp,"<span style=\"background-color:"+opts.highlightColor+"\">$1</span>") //disabling regex disables html matching too
+				continue;
+			}
+			el.innerHTML = el.innerHTML.replace(rexp,"<span style=\"background-color:"+opts.highlightColor+"\">$0</span>") //i cant figure out a proper, non snowflakey way to let people select the group that gets highlighted
```

Comment:
If anyone has a good solution to the problem of the user being able to select the capture group they want to highlight without adding another textbox to specify, please let me know.  I've talked with a few people and had came up with only the following:
1: Use $1 (Not preferred because backreferences can cause issue)
2: Use the last group (Mildly arbitrary and can be confusing)
3: Use the group with the longest match (Could be odd if a group not meant to be highlighted is longer than the one that is highlighted, but i cant think of a situation that would cause this, also mildly arbitrary and confusing)
4: Use the first group with the most identical matches (seemingly the best option here but a bit snowlakey and confusing, ex: `(a)((ab))` would match `aab` as `a,ab,ab` and we would select `ab` since its repeated)